### PR TITLE
DRIVERS-1868: Integration tests for observeSensitiveCommands

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/DRIVERS-1868

These tests pass locally in PHPLIB while working on [PHPLIB-664](https://jira.mongodb.org/browse/PHPLIB-664) and [PHPLIB-676](https://jira.mongodb.org/browse/PHPLIB-676) (actual tickets for command redaction).

### CODE REVIEW

**Review by alcaeus — APPROVED** [review]
LGTM, even if these tests have to duplicate some of the tests from the original command redaction tests.

**Comment by jmikola** [comment]
@prashantmital Did you get a chance to run this through PyMongo (per our conversation yesterday)?

**Comment by prashantmital** [comment]
@jmikola Yes I did. The tests fail on PyMongo with the following output:

```
======================================================================
FAIL: test_hello_with_speculativeAuthenticate (test.test_unified_format.TestUnifiedTestFormatValidPassObserveSensitiveCommands)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1136, in test_case
    self.run_scenario(spec)
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1123, in run_scenario
    self.check_events(spec.get('expectEvents', []))
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1067, in check_events
    self.assertEqual(actual_events, [])
AssertionError: Lists differ: [<CommandStartedEvent ('localhost', 27018)[219 chars]one>] != []

First list contains 2 additional elements.
First extra element 0:
<CommandStartedEvent ('localhost', 27018) db: 'observeSensitiveCommands', command: 'hello', operation_id: 2014119113, service_id: None>

+ []
- [<CommandStartedEvent ('localhost', 27018) db: 'observeSensitiveCommands', command: 'hello', operation_id: 2014119113, service_id: None>,
-  <CommandSucceededEvent ('localhost', 27018) command: 'hello', operation_id: 2014119113, duration_micros: 409, service_id: None>]

======================================================================
FAIL: test_legacy_hello_with_speculativeAuthenticate (test.test_unified_format.TestUnifiedTestFormatValidPassObserveSensitiveCommands)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1136, in test_case
    self.run_scenario(spec)
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1123, in run_scenario
    self.check_events(spec.get('expectEvents', []))
  File "/Users/pmital/Developer/mongo-python-driver/test/unified_format.py", line 1067, in check_events
    self.assertEqual(actual_events, [])
AssertionError: Lists differ: [<CommandStartedEvent ('localhost', 27018)[495 chars]one>] != []

First list contains 4 additional elements.
First extra element 0:
<CommandStartedEvent ('localhost', 27018) db: 'observeSensitiveCommands', command: 'ismaster', operation_id: 2049590396, service_id: None>

+ []
- [<CommandStartedEvent ('localhost', 27018) db: 'observeSensitiveCommands', command: 'ismaster', operation_id: 2049590396, service_id: None>,
-  <CommandSucceededEvent ('localhost', 27018) command: 'ismaster', operation_id: 2049590396, duration_micros: 872, service_id: None>,
-  <CommandStartedEvent ('localhost', 27018) db: 'observeSensitiveCommands', command: 'isMaster', operation_id: 621301815, service_id: None>,
-  <CommandSucceededEvent ('localhost', 27018) command: 'isMaster', operation_id: 621301815, duration_micros: 624, service_id: None>]
```

I _think_ this is because of how PyMongo performs redaction of commands. The CommandStarted/Succeeded events themselves are not suppressed. The command/response documents are instead replaced with blank documents for security sensitive commands. These tests seem to expect that no events are emitted for security sensitive commands?

**Comment by prashantmital** [comment]
On a closer look, PyMongo should pass these tests just fine... I'll investigate the reason for the failures

**Review by prashantmital — APPROVED** [review]
The issue in PyMongo was caused by our unified runner implementation. So this test actually just caught an issue with PyMongo's implementation of `observeSensitiveCommands`. LGTM!

**Comment by jmikola** [comment]
> So this test actually just caught an issue with PyMongo's implementation of `observeSensitiveCommands`

Glad to hear. That was the intention of this PR.

File: source/unified-test-format/unified-test-format.rst [hunk 1]
```diff
@@ -2779,7 +2782,13 @@ events for the following:
...
 - Any commands containing sensitive information (per the
   `Command Monitoring <../command-monitoring/command-monitoring.rst#security>`__
-  spec).
+  spec) unless
+  `observeSensitiveCommands <entity_client_observeSensitiveCommands_>`_ is true.
+  Note that drivers will redact commands and replies for sensitive commands. For
+  ``hello`` and legacy hello, which are conditionally sensistive based on the
+  presence of a ``speculativeAuthenticate`` field, the test runner may need to
+  infer that the events are sensitive based on whether or not the command and
+  reply documents are redacted (i.e. empty documents).
```
Comment (jmikola) [inline source/unified-test-format/unified-test-format.rst:2791]: @prashantmital: I just added this to capture the point made in my last PR comment. Let me know if this helps, as I expect other drivers may run into similar issues.

Comment (prashantmital) [reply source/unified-test-format/unified-test-format.rst:2791]: Yup, this is perfect... There is a distinction between command redaction and events being issued/captured which is obviously implied by the original language but made much more explicit with the addition of this new text. LGTM!

**Review by prashantmital — APPROVED** [review]

**Pull request merged**